astropy/utils/data_info.py

```diff
@@ -384,3 +384,8 @@ def iter_str_vals(self):
 
 class MixinInfo(BaseColumnInfo):
     pass
+
+class ParentDtypeInfo(MixinInfo):
```

Comment:
I think I'd make `QuantityInfo` a subclass of this one too (even though its definition would not change), just so a suitable `isinstance(info, ParentDtypeInfo)` would give the right answer.